lib/twom.h

```diff
@@ -0,0 +1,162 @@
+/* twom.h - twoskip implementation with MVCC capability
+ *
+ * Copyright (c) 2025 Fastmail Pty Ltd
+ *
+ * https://creativecommons.org/publicdomain/zero/1.0/
+ *
+ *   The person who associated a work with this deed has dedicated the work to the
+ *   public domain by waiving all of his or her rights to the work worldwide under
+ *   copyright law, including all related and neighboring rights, to the extent
+ *   allowed by law.
+ *
+ *   You can copy, modify, distribute and perform the work, even for commercial
+ *   purposes, all without asking permission.
+ */
+
+#ifndef INCLUDED_TWOM_H
+#define INCLUDED_TWOM_H
+
+#include <stdlib.h>
+#include <stdint.h>
+
+struct twom_db;
+struct twom_txn;
+struct twom_cursor;
+
+enum twom_ret {
+    TWOM_OK = 0,
+    TWOM_DONE = 1,
+    TWOM_IOERROR = -1,
+    TWOM_AGAIN = -2,
+    TWOM_EXISTS = -3,
+    TWOM_INTERNAL = -4,
+    TWOM_NOTFOUND = -5,
+    TWOM_LOCKED = -6,
+    TWOM_READONLY = -9,
+};
+
+// we don't reuse flags, there's 32 bits of space, and we spread through them
+enum twom_flagspec {
+    TWOM_CREATE          = 1<<0,    /* Create the database if not existant */
+    TWOM_SHARED          = 1<<1,    /* Open in shared lock mode */
+    TWOM_NOCSUM          = 1<<2,    /* Don't check checksums on read */
+    TWOM_NOSYNC          = 1<<3,    /* Don't msync/fsync on write */
+
+    TWOM_ALWAYSYIELD     = 1<<9,
```

Comment:
I can guess the purpose of most of these flags, but a one-liner docstring like for the first four flags would help.